7.10.0 release changelog (#930)
elastic/helm-charts: BREAKING_CHANGES.md

<!-- Use this to update TOC: -->
<!-- docker run --rm -it -v $(pwd):/usr/src jorgeandrada/doctoc --github -->

## 7.10.0

### Migration to Helm 3

> Starting from 7.10.0 release, Helm 3 is fully supported in Elastic Helm charts

Reviewer: ```suggestion
Starting from the 7.10.0 release, Helm 3 is fully supported in Elastic Helm charts
```

Error: UPGRADE FAILED: Deployment.apps "mb-metricbeat-metrics" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"mb-metricbeat-metrics", "release":"mb"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

Unfortunately using `helm upgrade --force` with Helm 3 is not enough. This chart
will need to be uninstalled and re-installed.

#### Elasticsearch upgrade with persistence.labels.enabled

If you are using `persistence.labels.enabled=true` with Elasticsearch, upgrade
will fail even with `--force`.

You'll need to deploy a new release with the same `clusterName` using Helm 3,
migrate your data, then remove the old release.

#### Rendered manifests contain a resource that already exists error

We experimented some `rendered manifests contain a resource that already exists`
errors with some charts upgrade, mostly for charts deploying `ClusterRole` and
`ClusterRoleBinding`resources.

Helm 3 is automatically adding some new annotations (`meta.helm.sh/release-name`
and `meta.helm.sh/release-namespace`) and labels (`app.kubernetes.io/managed-by`)

> to the charts resources during upgrade, however for some reason it seems to not

Reviewer: ```suggestion
to the charts resources during upgrade. However it sometimes fail to add the annotations with the below error:
```

> add sometimes and fails with the following error:

Reviewer: ```suggestion
```
